packages/ens-utils/src/registrar.test.ts

```diff
@@ -0,0 +1,77 @@
+import { describe, expect, it } from "vitest";
+import { scaleAnnualPrice } from "./registrar";
+import { buildPrice } from "./price";
+import { Currency } from "./currency";
+import { buildDuration, SECONDS_PER_YEAR } from "./time";
+
+describe("scaleAnnualPrice() function", (t) => {
+  it("should scale a one year price to a half an year price", () => {
+    const price = buildPrice(100n, Currency.Usd);
+    const years = 0.5;
+    const secondsPerYearAsNumber = Number(SECONDS_PER_YEAR.seconds);
+    const duration = BigInt(secondsPerYearAsNumber * years);
+    const secondsPerYears = buildDuration(duration);
+
+    const expectedResult = {
```

Comment:
@FrancoAguzzi Please avoid manually building prices. use `buildPrice`.

Additionally, in a context like this I think it's best to explicitly define the expected result rather than to dynamically compute it.